### ISSUE TITLE
psychopy.data.getDateStr() returns non-ASCII characters

### ISSUE THREAD

**Issue by hsogo** [psychopy/psychopy#1543]
The latest Builder in the repository cannot compile .py file.  My environment is Japanese Windows 10 + WinPython 2.7.10.  Error message is following:

> Traceback (most recent call last):
>   File "C:\Users\Hiroyuki\Dropbox\git-repository\psychopy\psychopy\app\builder\builder.py", line 1972, in runFile
>     script = self.generateScript(self.exp.expPath)
>   File "C:\Users\Hiroyuki\Dropbox\git-repository\psychopy\psychopy\app\builder\builder.py", line 2178, in generateScript
>     target=target)
>   File "C:\Users\Hiroyuki\Dropbox\git-repository\psychopy\psychopy\app\builder\experiment.py", line 231, in writeScript
>     self.psychopyVersion, localDateTime)
>   File "C:\Users\Hiroyuki\Dropbox\git-repository\psychopy\psychopy\app\builder\components\settings\\_\_init\_\_.py", line 277, in writeInitCode
>     'If you publish work using this script please cite the PsychoPy '
>   File "C:\Users\Hiroyuki\Dropbox\git-repository\psychopy\psychopy\app\builder\experiment.py", line 128, in write
>     io.StringIO.write(self, u"{}".format(text))
> UnicodeDecodeError: 'ascii' codec can't decode byte 0x8c in position 133: ordinal not in range(128)

I inserted `print(text)` before line 128 of app/builder/experiment.py and found that non-ASCII characters were included in the date of generating .py file.  This string is generated at writeInitCode() of SettingsComponent (line 276 of app/builder/components/settings/\_\_init\_\_.py).

```
    def writeInitCode(self, buff, version, localDateTime):

        buff.write(
            '#!/usr/bin/env python\n'
            '# -*- coding: utf-8 -*-\n'
            '"""\nThis experiment was created using PsychoPy2 Experiment '
            'Builder (v%s),\n'
            '    on %s\n' % (version, localDateTime) +                                            ### <- Here
            'If you publish work using this script please cite the PsychoPy '
            'publications:\n'
            '    Peirce, JW (2007) PsychoPy - Psychophysics software in '
            'Python.\n'
            '        Journal of Neuroscience Methods, 162(1-2), 8-13.\n'
            '    Peirce, JW (2009) Generating stimuli for neuroscience using '
            'PsychoPy.\n'
            '        Frontiers in Neuroinformatics, 2:10. doi: 10.3389/'
            'neuro.11.010.2008\n"""\n'
            "\nfrom __future__ import absolute_import, division\n")

        self.writeUseVersion(buff)
```

The value of 'IocalDateTime' is generated at line 227 of app/builder/experiment.py. 

```
            localDateTime = data.getDateStr(format="%B %d, %Y, at %H:%M") 
```

I thought that data.getDateStr() returns non-ASCII characters and run following script from PsychoPy Coder.

```
import psychopy.data as data
print(data.getDateStr(format="%B %d, %Y, at %H:%M"))
```

I expected that non-ASCII characters would be printed, but the output was like **'October 06, 2017, at 17:35'**.  Non-ASCII characters were not included.

Next, I inserted `print(localDateTime)` below line 227 of app/builder/experiment.py as following:

```
            localDateTime = data.getDateStr(format="%B %d, %Y, at %H:%M")   #line 227
            print(localDateTime)     # This line is inserted
```

Then I restarted PsychoPy Builder and compile experiment.  In this case, I got non-ASCII output like **'10月 06, 2017, at 17:13'** !
I confirmed that this error doesn't happen in PsychoPy 1.84.3b.

It seems that locale setting is changed somewhere, but I can't find where it occurs at present. 
I'd appreciate any suggestions.

**Comment by hsogo** [comment]
Ah, I understand what happened!
In 1.85.3b, date string was attempted to converted to unicode. 

```
def getDateStr(format="%Y_%b_%d_%H%M"):
    """Uses ``time.strftime()``_ to generate a string of the form
    2012_Apr_19_1531 for 19th April 3.31pm, 2012.
    This is often useful appended to data filenames to provide unique names.
    To include the year: getDateStr(format="%Y_%b_%d_%H%M")
    returns '2011_Mar_16_1307' depending on locale, can have unicode chars
    in month names, so utf_8_decode them
    For date in the format of the current localization, do:
        data.getDateStr(format=locale.nl_langinfo(locale.D_T_FMT))
    """
    now = time.strftime(format, time.localtime())
    try:
        now_decoded = codecs.utf_8_decode(now)[0]
    except UnicodeDecodeError:
        # '2011_03_16_1307'
        now_decoded = time.strftime("%Y_%m_%d_%H%M", time.localtime())

    return now_decoded
```

On the other hand, in the latest version, getDataStr() simply returns the returned value of time.strftime().

```
def getDateStr(format="%Y_%b_%d_%H%M"):
    """Uses ``time.strftime()``_ to generate a string of the form
    2012_Apr_19_1531 for 19th April 3.31pm, 2012.
    This is often useful appended to data filenames to provide unique names.
    To include the year: getDateStr(format="%Y_%b_%d_%H%M")
    returns '2011_Mar_16_1307' depending on locale, can have unicode chars
    in month names, so utf_8_decode them
    For date in the format of the current localization, do:
        data.getDateStr(format=locale.nl_langinfo(locale.D_T_FMT))
    """
    return time.strftime(format, time.localtime())
```

**Comment by hoechenberger** [comment]
Closed by #1544.
